org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/XbaseCompiler.java

```diff
@@ -474,11 +518,17 @@ protected void _toJavaStatement(XTryCatchFinallyExpression expr, ITreeAppendable
 			b.append(";");
 		}
 		b.decreaseIndentation().newLine().append("}");
-		appendCatchAndFinally(expr, b, isReferenced);
+
+		// Catch and Finally
+		appendCatchAndFinally(expr, b, isReferenced, java7);
 	}
 
-	protected void appendCatchAndFinally(XTryCatchFinallyExpression expr, ITreeAppendable b, boolean isReferenced) {
+	protected void appendCatchAndFinally(XTryCatchFinallyExpression expr, ITreeAppendable b, boolean isReferenced, boolean java7) {
```

Comment:
Unfortunately we cannot change the signature of this method to stay backwards compatible

Reply:
I wonder if we should drop support for older targets anyway

Reply:
it's changed, it is not necessary to pass that as a parameter.